core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -219,7 +219,10 @@ final class Stream[+F[_],+O] private (private val coreRef: Stream.CoreRef[F,O])
   def pure(implicit S: Sub1[F,Pure]): Stream[Pure,O] = covary[Pure]
 
   /** Repeat this stream an infinite number of times. `s.repeat == s ++ s ++ s ++ ...` */
-  def repeat: Stream[F,O] = self ++ repeat
+  def repeat: Stream[F,O] = for {
+    a <- self.uncons1
+    b <- a.map { case (h, t) => t.cons1(h) ++ repeat }.getOrElse(Stream.empty)
```

Comment:
What is the value in special-casing like this?  I don't mean to be negative, but it seems like a lot of extra awkwardness just to identify the empty stream in one specific case.